File: benchmarks/tasks/py-agent-escalation/repo/src/agent/decision.py
from src.shared.agent import AgentDecision, is_agent_decision


def parse_agent_decision(value: object) -> AgentDecision:
    if not is_agent_decision(value):
        raise ValueError("Invalid agent decision")

    return value


def should_escalate(decision: AgentDecision) -> bool:
    return decision["action"] == "ask_followup" and decision["confidence"] < 0.5

File: .clasp-loops/swarm-ready-full-goal-live/baseline-workspace/benchmarks/tasks/ts-agent-escalation/repo/src/agent/decision.ts
import { AgentDecision, isAgentDecision } from "../shared/agent.js";

export function parseAgentDecision(value: unknown): AgentDecision {
  if (!isAgentDecision(value)) {
    throw new Error("Invalid agent decision");
  }

  return value;
}

export function shouldEscalate(decision: AgentDecision): boolean {
  return decision.action === "ask_followup" && decision.confidence < 0.5;
}

